engine/src/test/java/io/camunda/zeebe/engine/processing/message/MessageStartEventSubscriptionTest.java

```diff
@@ -280,13 +282,35 @@ public void shouldHaveSameSubscriptionKey() {
         .create();
 
     assertThat(
-            RecordingExporter.messageSubscriptionRecords(MessageSubscriptionIntent.CREATED)
-                .withMessageName("event_message")
-                .withCorrelationKey("key")
-                .findAny())
+        RecordingExporter.messageSubscriptionRecords(MessageSubscriptionIntent.CREATED)
+            .withMessageName("event_message")
+            .withCorrelationKey("key")
+            .findAny())
         .isPresent();
   }
 
+  @Test
+  public void shouldOpenSingleMessageSubscriptionOnMultipleDeployments() {
+    // give
+    final var process = createProcessWithOneMessageStartEvent();
+
+    engine.deployment().withXmlResource(process).deploy();
+
+    // when
+    engine.deployment().withXmlResource(process).deploy();
+
+    // then
+    assertThat(RecordingExporter.messageStartEventSubscriptionRecords(
+            MessageStartEventSubscriptionIntent.CREATED)
+        .withBpmnProcessId("processId")
+        .limit(2))
+        .extracting(Record::getValue)
+        .extracting(MessageStartEventSubscriptionRecordValue::getMessageName,
+            MessageStartEventSubscriptionRecordValue::getStartEventId)
+        .hasSize(1)
+        .containsExactly(tuple(MESSAGE_NAME1, EVENT_ID1));
```

Comment:
❌ Since the fix, this `limit(2)` will never hit the second record because only one such record is recorded. This means the test will be very slow because it waits for the record stream to time out (5 seconds).

🤔 A fundamental question pops up when attempting to resolve this problem. How can we limit the recorded stream to a record that does not exist on the stream? I.e., the second subscription record does not exist on the stream, so how else can we limit the log? 

💡 There is no good answer, but there is a trick we could try. By adding another command that will be rejected (after the command that we want to test), we can use the position of the command to limit the stream. That means we now have all the records on the stream from the start to the end, plus the rejected command we added additionally. Of course, this makes the test worse. However, sometimes that may be a cost we can pay when it helps us avoid a bug. That would look something like this:

```java
// when
engine.deployment().withXmlResource(process).deploy();

// then
// deploy an empty deployment, so we can use the position to limit the recorded stream
final var position = engine.deployment().expectRejection().deploy().getPosition();
assertThat(
        RecordingExporter.records()
            .limit(r -> r.getPosition() >= position)
            .filter(r -> r.getValueType() == ValueType.MESSAGE_START_EVENT_SUBSCRIPTION))
    .describedAs("Expect only one message start event subscription for duplicate deployments")
    .hasSize(1);
```